python/llm/scripts/ipex-llm-init

```diff
@@ -88,94 +103,68 @@ disable_tcmalloc
 LD_PRELOAD=""
 
 OPTIND=1
+DEVICE=""
 
-while getopts "hojtcgd:-:" opt; do
-    case ${opt} in
-        - )
-            case "${OPTARG}" in
-                help)
-                    display-help
-                    return 0
-                    ;;
-                gomp)
-                    disable_iomp
-                    ;;
-                jemalloc)
-                    enable_jemalloc
-                    ;;
-                tcmalloc)
-                    enable_tcmalloc
-                    ;;
-                disable-allocator)
-                    disable_jemalloc
-                    disable_tcmalloc
-                    ;;
-                gpu)
-                    enable_gpu
-                    ;;
-                debug)
-                    display-var
-                    return 0
-                    ;;
-                *)
-                    display-error $OPTARG
-                    return 1
-                    ;;
-            esac
-            ;;
-
-        h )
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h|--help)
             display-help
             return 0
             ;;
-        o )
+        -o|--gomp)
             disable_iomp
+            shift
             ;;
-        j )
+        -j|--jemalloc)
             enable_jemalloc
+            shift
             ;;
-        t )
+        -t|--tcmalloc)
             enable_tcmalloc
+            shift
             ;;
-        c )
+        -c|--disable-allocator)
             disable_jemalloc
             disable_tcmalloc
+            shift
             ;;
-        g )
+        -g|--gpu)
             enable_gpu
+            shift
             ;;
-        d )
+        -d|--debug)
             display-var
             return 0
             ;;
-        \? )
-            display-error $OPTARG
+        --device)
+            DEVICE="$2"
+            shift 2
+            ;;
+        *)
+            display-error "$1"
             return 1
             ;;
     esac
 done
 
-shift $((OPTIND -1))
-
 # Find ipex-llm-init dir
 if [ ! -z $BASH_SOURCE ]; then
     # using bash
     if [ "$BASH_SOURCE" = "$0" ]; then
         echo "Error: Incorrect usage: ipex-llm-init must be sourced."
-        exit 1
```

Comment:
`return` can only be used in a function or sourced script.

This code is checking whether user is sourcing script or directly executing script, if use `return` and user is executing, it will report an error like `return: can only return from a function or sourced script`, not exits with error code 1, so we should use `exit 1` here.

others LGTM